packages/eslint-plugin-ckeditor5-rules/lib/rules/ckeditor-plugin-flags.js

```diff
@@ -0,0 +1,680 @@
+				messageId: 'definedBeforePluginName',
+				data: {
+					flag: requiredFlag.name
+				}
+			} );
+		}
+	}
+
+	// Return plain list of methods that are in the wrong order to ignore them in the next check.
+	return flagsDefinedBeforePluginName.flatMap( ( [ , methodsSomewhereInClass ] ) => methodsSomewhereInClass );
+}
+
+/**
+ * Checks if all required flags are defined in the proper order.
+ *
+ * @param {Object} context The ESLint rule context.
+ * @param {Object} classDeclaration The class declaration node to analyze.
+ * @param {Array.<Object>} ignoredMethods A list of methods that should be ignored during the check.
+ * @example
+ *
+ * Let's assume we have the following class declaration:
+ *
+ * ```
+ * class TestPlugin extends Plugin {
+ * 	static get pluginName() { return 'TestPlugin'; }
+ * 	static get isOfficialPlugin() { return true; }
+ * 	constructor() { super(); }
+ * 	static get isPremiumPlugin() { return false; }
+ * }
+ * ```
+ *
+ * This check will report an error for the `isPremiumPlugin` method because it
+ * should be defined directly after the `isOfficialPlugin` method.
+ */
+function checkIfFlagsDefinedInProperOrder( context, classDeclaration, ignoredMethods = [] ) {
+	const orderedMethodsLinkedList = getRequiredFlagsConfig( context ).map( ( flag, index, array ) => ( {
+		name: flag.name,
+		before: {
+			name: array[ index - 1 ]?.name || PLUGIN_NAME_METHOD_NAME
+		}
+	} ) );
+
+	let lastGetterMethod = null;
+
+	for ( const classNode of getAllClassMembers( classDeclaration ) ) {
+		// Ignore if method signature is not a getter or it's not a static method.
+		if ( !isStaticGetterMethod( classNode ) ) {
+			lastGetterMethod = null;
+			continue;
+		}
+
+		// Check if method was reported by one of previous analyzers. If so, ignore it, do not report similar error twice.
+		if ( ignoredMethods.includes( classNode ) ) {
+			continue;
+		}
+
+		// Ignore if method is not a flag method.
+		const flagMethod = orderedMethodsLinkedList.find( element => element.name === classNode.key.name );
+
+		if ( flagMethod ) {
+			if ( !lastGetterMethod ) {
+				// If there is no last method, it means that the current method is the first flag method after
+				// constructor or any other non-flag method. It's bad.
+				context.report( {
+					node: classNode,
+					fix: createPluginsFlagsFixer( context, classDeclaration ),
+					messageId: 'definedInProperOrder',
+					data: {
+						flag: classNode.key.name
+					}
+				} );
+			} else if ( lastGetterMethod && lastGetterMethod.key.name !== flagMethod.before.name ) {
+				// If the last flag method is not the one that should be before the current flag method, report an error.
+				context.report( {
+					node: classNode,
+					fix: createPluginsFlagsFixer( context, classDeclaration ),
+					messageId: 'definedAfterSpecificMethod',
+					data: {
+						flag: classNode.key.name,
+						before: flagMethod.before.name
+					}
+				} );
+			}
+		}
+
+		lastGetterMethod = classNode;
+	}
+}
```

Comment:
I'd remove it. We have a code style guide and we should refer to it without strict enforcing as there might be some exceptions.